# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: serkan@gentoo.org-20110313222348-t01i70f8rcedd8by
# target_branch: ../mainline
# testament_sha1: 131e1368d563e766440b56049743d2dc2f9ea6e1
# timestamp: 2011-03-14 00:28:45 +0200
# source_branch: /usr/portage/dev-java/java-gnome/files/java-gnome-\
#   4.0.19-libnotify-0.7.patch
# base_revision_id: andrew@operationaldynamics.com-20110214064042-\
#   fv6tgq41rru7wjng
# 
# Begin patch
=== modified file 'configure'
--- configure	2011-02-14 06:40:42 +0000
+++ configure	2011-02-14 22:27:29 +0000
@@ -805,7 +805,7 @@
 			"gtkspell");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify");
 
@@ -853,7 +853,7 @@
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 
@@ -917,7 +917,7 @@
 			"libgtksourceview2.0-dev");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify-dev");
 
@@ -968,7 +968,7 @@
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 
@@ -1020,7 +1020,7 @@
 			"lib${cpu_arch}gtksourceview-2.0-devel");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"lib${cpu_arch}notify-devel");
 
@@ -1083,7 +1083,7 @@
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 

=== modified file 'src/bindings/org/gnome/notify/Notification.java'
--- src/bindings/org/gnome/notify/Notification.java	2011-02-14 03:34:07 +0000
+++ src/bindings/org/gnome/notify/Notification.java	2011-03-13 22:23:48 +0000
@@ -80,41 +80,7 @@
      */
 
     public Notification(String summary, String body, String icon) {
-        super(NotifyNotification.createNotification(summary, body, icon, null));
-    }
-
-    /**
-     * Create a new notification.
-     * 
-     * <p>
-     * The summary appears on the titlebar of notification and body appears as
-     * its text. Icon may be a string defining a theme icon or the filename
-     * identifying the icon that appears next to text. Attach identifies the
-     * widget that the notification relates to.
-     * 
-     * <p>
-     * Note that all but summary can be <code>null</code>.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-
-    public Notification(String summary, String body, String icon, Widget attach) {
-        super(NotifyNotification.createNotification(summary, body, icon, attach));
-    }
-
-    /**
-     * Create a new notification attached to a {@link StatusIcon}. See
-     * {@link #Notification(String,String,String,Widget) Notification()} for
-     * other parameters.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-    public Notification(String summary, String body, String icon, StatusIcon statusIcon) {
-        super(NotifyNotification.createNotificationWithStatusIcon(summary, body, icon, statusIcon));
+        super(NotifyNotification.createNotification(summary, body, icon));
     }
 
     /**
@@ -131,39 +97,6 @@
     }
 
     /**
-     * Attaches Notification to a Widget setting hints to its location.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-    public void attach(Widget attach) {
-        NotifyNotification.attachToWidget(this, attach);
-    }
-
-    /**
-     * Attaches Notification to a {@link StatusIcon} setting hints to its
-     * location.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-    public void attach(StatusIcon statusIcon) {
-        NotifyNotification.attachToStatusIcon(this, statusIcon);
-    }
-
-    /**
-     * Sets the position of the notification to display on screen.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3
-     */
-    public void setGeometryHints(Screen screen, int x, int y) {
-        NotifyNotification.setGeometryHints(this, screen, x, y);
-    }
-
-    /**
      * Display the notification on screen.
      * 
      * @since 4.0.12

=== modified file 'src/defs/NotifyNotification.defs'
--- src/defs/NotifyNotification.defs	2009-05-16 05:46:28 +0000
+++ src/defs/NotifyNotification.defs	2011-01-31 17:58:10 +0000
@@ -15,19 +15,6 @@
     '("const-gchar*" "summary")
     '("const-gchar*" "body" (null-ok))
     '("const-gchar*" "icon" (null-ok))
-    '("GtkWidget*" "attach" (null-ok))
-  )
-)
-
-(define-function notify_notification_new_with_status_icon
-  (is-constructor-of "NotifyNotification")
-  (c-name "notify_notification_new_with_status_icon")
-  (return-type "NotifyNotification*")
-  (parameters
-    '("const-gchar*" "summary")
-    '("const-gchar*" "body" (null-ok))
-    '("const-gchar*" "icon" (null-ok))
-    '("GtkStatusIcon*" "status_icon")
   )
 )
 
@@ -42,35 +29,6 @@
   )
 )
 
-(define-method attach_to_widget
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_widget")
-  (return-type "none")
-  (parameters
-    '("GtkWidget*" "attach")
-  )
-)
-
-(define-method attach_to_status_icon
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_status_icon")
-  (return-type "none")
-  (parameters
-    '("GtkStatusIcon*" "status_icon")
-  )
-)
-
-(define-method set_geometry_hints
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_set_geometry_hints")
-  (return-type "none")
-  (parameters
-    '("GdkScreen*" "screen")
-    '("gint" "x")
-    '("gint" "y")
-  )
-)
-
 (define-method show
   (of-object "NotifyNotification")
   (c-name "notify_notification_show")

# Begin bundle
IyBCYXphYXIgcmV2aXNpb24gYnVuZGxlIHY0CiMKQlpoOTFBWSZTWWU6Q8IACDNfgGQwUH///3//
/3C////4YAxdXRdzx6vvbaqzaGw+7dtp7Punrt7u4U5QOgK6oASSmlT2ptNBMo9NoyDKR5TynpjV
PKbU9DKHqeoyAPUCSkwjJiNJhTaqAAaMjIBhAAHqaM1AZJPJpNSeoaeoyaAGgAAMgAAAACRIEE01
T8kxKPKPSeo9TNMRGg0yAaDIyABFIRGmg1NNDJo0NGiankanpk1P1ExGJkABoJKBGhoJiMQ0nqU9
R71UNGgNANDIDQA03FckDaB2FYFoLimny8T3jVhugOZ8Ca9V+3mo+ac83NRU78arwQs3+4YMNtQw
/3oQce76Va8EbWb9XpLLkdvmhaIKZxq9jxZjSJOTq8WZMJZayHeYSAMJzcJCUFpDxCCSSMUNEmVT
Q5i9Fq3bZmzCBI02BPpF+QWUm4Ahg2mxttNo9vYLb6wD2r/Kq1zhQNtDhm+Dvbom4lLknnLV0mY2
mWmRcTi1LVd051lz10fdxMLYyZKB0IlCwaJ3xgy1LSpWAxeIVZoHmjNFqUkvkbINJrOSbKHoBU9J
A0xjesMprRyMCC89F2ujH5Xe760JC7u64VzEFFRYh33AtWu+1OySYsV7EVR9OwFgflT/1wDNhUmM
aRvznsRPEAsaFSg6HIVAmEGVSW3mgWR46s3ddlFVDWJC3oOxA4N0/vlCQcXSVPGOvYMCEDCpoELz
6AgejknrTfHE0U5Fa3S9Tw5GAKgi9WNj0YkqrkDl3YExI9KKWah73t/3o1/pLLM4LZyCUpMpS5rc
/Xe0Zp+J8eBqkFb4OE4AbFDSwX1CYgahz6VJJyL1Vd8ElAy5JKpSgDh0gUZqlI0hhJLl1nNIwYGx
Ite0R5ekii1zZiXb8r7WRbBGJ2GbuAxskjOuljOBUCpHOIgmpTbUFHUQMtLFTSMk3MmpqGGsQOto
cwD0tSUuor11c0RTjK5ojd0Qzc1pOCbCtxn4FQQWq0ePEQSaEOjHiHoHgPUzGQ9kXXzLDvSVs4At
ZWSvEHN9SkoomxCqWZs/nqwbPf16OEy4zfGc8RA5BssplWMiK4FBxQx6Sb/q4LAmKUhhEj3ldpmk
SvvGE87KNshB9CjCPe3dDHe60cpZp1lATRQKfYN3GjOGmfNSaAHaSNS3i6YtdOUjJoOYYPZEij/l
87kBxTEIXCrTPrTeiEThlmOhu8CCSaA0KWvckoXPaBbne8mEJUGygREOziQZ7O3CVagHMrtL8Ai0
LbdtSUjSZzSskMAz6EihqTxOJUkQX5ynsJeTPIQVYhTPTFNgVjZPGRUoJGN7lxovHvEHE3PSDVQ6
XoJbupfMzMQL4jWw4oYWPZhPTCqd3sFbCw0pKYTBGCYQGVl0ba4lt0nxKNxByDkHvBqkbziDqQTI
4WCxIJzgfohiOOLQXFkHtV0kNzRW1JQaXmNpQwxJ5nxkNsM1GrIcMe0fMBURjK0Jnl3glaxDDMuF
egyfFzCuq8g4tG6eGOR4lyj4WmsfXhlyM4msb9lZzKA8ZEqkJFYvYXIftaN09C4uSHcPN+CFCgXR
3M2kzTRpq7KIgqOwzLAztqVOjulBlolw2eObOOH55GLCt2c3uNijXuWDSbHDUTpPUqy1079kcCqS
cjLY9CxskZ9Oy+xnNuhhe0xH20jfUvnJowWJjKy6COMfAuwoUNQlYFgQUYmFGphL+jdIep7k1oFy
uCirAjBYXQQzOFqNwNHJE0XrCSS2rzGD7G1xG3tCkQT4akPsjwwuZE88bhpcElyuUlo7Wpk8vyID
9CgzaMpV+8A1ks92mJJpsz2agMpsaggQbc9PdMC1m6ZRiUEKRhWSoC5xP1C4LqnsbPNaMIlz0w7d
rrC3xqkMs+RQVuFDTt0NS9BlByrI7Ckh3OpI2OgGWGmboZaQRtlvokskc5oX8hmkWcYOTgsZkGET
FAgTbAiRxvloXvGaSLieDakBPkQrYoNfViItwJX2VUkxd6GYyhhYlTEkW+BhHppYwzcHiCB2L3GW
NxYncW6dQQpOTFrDRwjFbuDJCjJWHbZRZJVQXAF0k16PMg4L0i+dd8MRjGGZEEKGlomxtyLLoPUD
ZeLtVkJHP1pGv293sXsWP579v4/acnUe8XetClwu+8d1gZhhkyRrZ5mY4iiD8OcA8HfeqpvE/Qv0
TFbuWTVgmHziyXpWioeXw68zX0u269/JZ41EKFB+l5kF7Mtk6uBZtOQXRZcT/lPH7t1eJ6sm8dLX
KmpVMmnNCrUvbdFDbFeTCgm3GtX2xa8hpLZCgr+JSQb3aVYrT8ReK7z1jc+i5qsE/RQTjLDEwyIi
hrFHYvGwKZbc66VCp9i44NFI3HWMyZNfTAHlwfBfisEHk8u80GUsC9A0EdRt8dbuQu8MihKKZuw7
3jZNkPnY8iemEfWq6eth6g/D8lgIMTt8Zjw82ui+AA6o4lPCBtYUh7gBo9qm/26QuAclp8GOaFzL
24UlUSKy6kjkgh3zKVjtvg8dZefcTlKVqhMSEGvwnCpgPoKVX2BjBzXKssLHYsZpvdp3i8QNhvjs
kali+gJwJCEQ3dXgSHWQH4vPhqwmMCYzONR6TkP014JKzXJgHQ1UObFn+YLFc3rkbjCDkjVI4A4o
Ou6ryOeY1a0FuhWvlz7a9pVZCLklac04X00galCgViD0RC282MAGFhlpRngu9BekftRNdSR19UC8
Pb8Ry/eQDQOibG+PA8F9t0eQiOPcZHj7EHc2mLqHe3S6bPuLhnN5wNScXkFiQWyPKmMTBrPU54rQ
Cjp3c5UaGJUdem2jQJXZHHrTTZ1V/KLlxvUISs0mU8dbZ24rnguyzcM5/m0H1buphDHmRTQhNXTT
Jqdyinw8/uYRT6tHTvnS4dwQucLATQlJbjHpBmkQXl2It28N+htc6GzTPebu2e0CBNb7MtlfhFuC
WswTLbnJchFOABwj/w945xnC+ZjgDATH1pfUK10iJt2BiDbR8/f4hPRYQQQAR9UUwBwssLm1pIeM
Qyz5nZ2PUzhEob/LOcPCgSDnAsbqOT6T05vSDyKyo6mpTUPxVnJ78brQgQLnAOnskIUG/xAG5jez
4+l6XQ1tjiGEYVhTvEPDBmfQzV23hz7OryB2oXcKlNDJLDQkpw4LvdJoVoAXn9YTAGnVRnygrggz
sh34HZ+tclIB8xA9PsHnZUq52wcIfPCO6+CO/0XjZnmVrahjbnJDwBoqUvP7WmJ3ipwIT6/fyYa2
oF5DWIZ5oURS3HMW4bVMrW4ACFKmzW3vGg1hRj8ExddG06HU3KQyj15N5QC2FeNkD3cWRFNQfZqx
G/LQgWMEIp7mQ2IpTapDsmAdqKYhLXnohj+T4kBAJBB0HMpoVk/Pl6GBhXGUQtvYBZ5O6rervvX/
PZaIV5UTyeyxaDZnop2JmTIpfLlTAH50yRJqAtoJopMIDnDOM0NqmFFLTwwuaYA0Enrq4nh9SZXL
tsF6kn30R6ZKoxTyoVNUjXbsVFqdy17ghTWGc2RMYZzJT2uNkt0QRvOYdVjN7UMrCFBbRasG0N7a
MQczQYAF/n/6VQ8+ZC+BfT6wkyygctbQaA8pnGQ0ucYBsHyTemZa3VkYVcMG4NFH1XqkUxAEIWfJ
lZCvjR7rGfMpWmYNl9nLTu31M97nHu8rjH71JHcOByGR3X22ptzNyFghnL6GgQ1+26+rmFflWpw6
1JN33q7ydwhdWmCA52Hs+0JDptPZPMp4q83D9LwJYGlamRHNETWTOG5Jn5B5VekpbfUkdKvOHvz3
IcSuRTOEwzjBE3RfsVvq8P/i7kinChIMp0h4QA==
